### PR TITLE
Initial plugin with tests

### PR DESCRIPTION
**Pull request by sc0ttkclark**
* Initial plugin: https://central.tri.be/issues/105892
* Shortcode: https://central.tri.be/issues/105893
* oEmbed provider: https://central.tri.be/issues/105911
* Tests: https://central.tri.be/issues/106979

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/**
+ * Main Tribe Loxi class.
+ */
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+if ( ! class_exists( 'Tribe__Loxi__Main' ) ) {
```
Comment (mitogh) [inline src/Tribe/Main.php]: I think returning early here will remove an indentation level or maybe to combine this one with the one above.

```php
if ( ! defined( 'ABSPATH' ) || class_exists( 'Tribe__Loxi__Main' ) )
```

File: src/Tribe/Main.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		 */
+		public function init() {
+
+			require_once TRIBE_LOXI_PLUGIN_DIR . '/src/Tribe/Shortcode.php';
+
+			// Register shortcode.
+			Tribe__Loxi__Shortcode::register();
+
+			// Get oembed loxi domain.
+			$loxi_domain = 'loxi.io';
```
Comment (mitogh) [inline src/Tribe/Main.php]: Maybe be good idea to save in a constant?

File: src/Tribe/Main.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+			// Set special domain if we are testing.
+			if ( defined( 'TRIBE_LOXI_SUBDOMAIN' ) && TRIBE_LOXI_SUBDOMAIN ) {
+				$loxi_domain = sanitize_key( TRIBE_LOXI_SUBDOMAIN ) . '.loxi.io';
```
Comment (mitogh) [inline src/Tribe/Main.php]: Maybe good idea to use the constant above here as well.

File: src/Tribe/Main.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+if ( ! class_exists( 'Tribe__Loxi__Main' ) ) {
+
+	/**
+	 * Main Tribe Loxi class.
```
Comment (mitogh) [inline src/Tribe/Main.php]: Missing `@since`

File: src/Tribe/Shortcode.php [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+ * Loxi Shortcode Class.
+ */
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+/**
+ * Loxi Shortcode Class.
```
Comment (mitogh) [inline src/Tribe/Shortcode.php]: Missing `@since`

File: src/Tribe/Shortcode.php [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+			$attributes['subdomain'] = $attributes['calendar'];
+
+			unset( $attributes['calendar'] );
+		}
+
+		if ( empty( $attributes['subdomain'] ) ) {
+			return '';
+		}
+
+		$loxi_domain = 'loxi.io';
```
Comment (mitogh) [inline src/Tribe/Shortcode.php]: Maybe we can call the const above here

**Review by mitogh — DISMISSED** [review]
Few questions comments

File: src/Tribe/Main.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Main.php]: I feel like I've seen some weirdness with doing class_exists checks at the top and not wrapping. Maybe it was only for functions. I'll test it out.

Comment (sc0ttkclark) [reply src/Tribe/Main.php]: No I was right, you have to wrap class definitions. I took it out of this file and moved it into the plugin init instead.

File: src/Tribe/Main.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/Tribe/Main.php]: Updated

File: src/Tribe/Main.php (hunk 3 above)
Comment (sc0ttkclark) [reply src/Tribe/Main.php]: Updated

File: src/Tribe/Main.php (hunk 4 above)
Comment (sc0ttkclark) [reply src/Tribe/Main.php]: Updated

File: src/Tribe/Shortcode.php (hunk 5 above)
Comment (sc0ttkclark) [reply src/Tribe/Shortcode.php]: Updated

File: src/Tribe/Shortcode.php (hunk 6 above)
Comment (sc0ttkclark) [reply src/Tribe/Shortcode.php]: Updated

**Review by mitogh — DISMISSED** [review]
👍👌

**Pull request merged**